tests/test_master.py

```diff
@@ -37,6 +43,75 @@ def test_raise_exception_when_task_timeout():
         master.column_names(client, None, tries=TRIES)
 
 
+def test_master_corr_matrix_is_combined_corr_matrix():
+    client = create_base_mock_client()
+
+    df1 = pd.DataFrame({'id': [1], COLUMN1: [123]})
+    df2 = pd.DataFrame({'id': [1], COLUMN2: [321]})
+
+    node_results = [df1, df2]
+    client.get_results.return_value = node_results
+
+    result = master.correlation_matrix(client, None, tries=TRIES, keys='id')
+    target = pd.DataFrame({'id': [1], COLUMN1: [123], COLUMN2: [321]}).corr()
+
+    pd.testing.assert_frame_equal(target, result)
+
+
+def test_master_corr_matrix_joins_on_multiple_keys():
+    client = create_base_mock_client()
+
+    df1 = pd.DataFrame(data=[KEY_VALUES + [123]], columns=KEYS + [COLUMN1])
+    df2 = pd.DataFrame(data=[KEY_VALUES + [321]], columns=KEYS + [COLUMN2])
+
+    client.get_results.return_value = [df1, df2]
+
+    target = pd.DataFrame(data=[KEY_VALUES + [123, 321]], columns=KEYS + [COLUMN1, COLUMN2]).corr()
+    result = master.correlation_matrix(client, None, keys=KEYS, tries=TRIES)
+
+    pd.testing.assert_frame_equal(target, result)
+
+
+def test_if_no_keys_provided_infer_keys():
```

Comment:
I cannot infer from the title what function is tested ;)

Reply:
Same goes for the following test function.

Reply:
added function name